src/main/java/com/synopsys/integration/blackduck/service/BlackDuckResponsesTransformer.java

```diff
@@ -59,12 +64,14 @@ public BlackDuckResponsesTransformer(BlackDuckHttpClient blackDuckHttpClient, Bl
             allResponses.addAll(blackDuckPageResponse.getItems());
 
             totalCount = blackDuckPageResponse.getTotalCount();
-            if (!getAll) {
+            if (maxPageCount <= 1) {
```

Comment:
This check doesn't seem necessary. Since you have the loop below, your loop should finish after its first iteration. Unless I'm missing something here.

Reply:
We always make a call for the first page first. We won't know the total number of items available to page until we make 1 request

Reply:
Perhaps we can make the loop smarter, but I didn't want to make too many changes to the behavior of this method.